tests/integration_tests/functional/test_cpu_features_aarch64.py

```diff
@@ -0,0 +1,145 @@
+# SPDX-License-Identifier: Apache-2.0
+"""Tests for the CPU features for aarch64."""
+
+# pylint: disable=too-many-lines
+
+import os
+import platform
+import re
+import shutil
+import sys
+import time
+from difflib import unified_diff
+from pathlib import Path
+
+import pandas as pd
+import pytest
+
+import framework.utils_cpuid as cpuid_utils
+from framework import utils
+from framework.artifacts import NetIfaceConfig
+from framework.defs import SUPPORTED_HOST_KERNELS
+from framework.utils_cpu_templates import SUPPORTED_CPU_TEMPLATES
+
+PLATFORM = platform.machine()
+UNSUPPORTED_HOST_KERNEL = (
+    utils.get_kernel_version(level=1) not in SUPPORTED_HOST_KERNELS
+)
+
+DEFAULT_G2_FEATURES = (
+    "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp "
+    "asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs"
+)
+
+DEFAULT_G2_FEATURES_NO_SSBS = (
+    "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp "
+    "asimdhp cpuid asimdrdm lrcpc dcpop asimddp"
+)
+
+DEFAULT_G3_FEATURES = (
+    "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp "
+    "asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp "
+    "sha512 asimdfhm dit uscat ilrcpc flagm ssbs"
+)
+
+DEFAULT_G3_FEATURES_NO_SSBS = (
+    "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp "
+    "asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp "
+    "sha512 asimdfhm dit uscat ilrcpc flagm"
+)
+
+DEFAULT_G3_FEATURES_V1N1 = (
+    "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp "
+    "asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs"
+)
+
+
+def _check_cpu_features_arm(test_microvm, template_name=None):
+    expected_cpu_features = {"Flags": []}
+    match cpuid_utils.get_instance_type():
+        case "m6g.metal":
+            match template_name:
+                case "aarch64_remove_ssbs":
+                    expected_cpu_features["Flags"] = DEFAULT_G2_FEATURES_NO_SSBS
+                case None:
+                    expected_cpu_features["Flags"] = DEFAULT_G2_FEATURES
+        case "c7g.metal":
+            match template_name:
+                case "aarch64_remove_ssbs":
+                    expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES_NO_SSBS
+                case "aarch64_v1n1":
+                    expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES_V1N1
+                case None:
+                    expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES
+
+    cpuid_utils.check_guest_cpuid_output(
+        test_microvm, "lscpu", None, ":", expected_cpu_features
+    )
+
+
+def get_cpu_template_dir(cpu_template):
+    """
+    Utility function to return a valid string which will be used as
+    name of the directory where snapshot artifacts are stored during
+    snapshot test and loaded from during restore test.
+
+    """
+    return cpu_template if cpu_template else "none"
+
+
+@pytest.mark.skipif(
+    PLATFORM != "aarch64",
+    reason="This is aarch64 specific test.",
+)
```

Comment:
This should be at the top of the file instead:

```python
pytestmark = pytest.mark.skipif(PLATFORM != "aarch64", "This is aarch64 specific test")
```

Reply:
Personally I like it being just above the test, seems more visible to me.